### PR TITLE
Improve PreTrainedModelWrapper._get_current_device

### PR DESCRIPTION
**Pull request by billvsme**
If using ```dummy_accelerator = Accelerator()```, it will cause the ```ACCELERATE_USE_DEEPSPEED``` environment variable to be set in advance, resulting in the regeneration of ```DeepSpeedPlugin``` and causing a change in the type of ```trainer.accelerator.state.deepspeed_plugin.hf_ds_config```
 [source code: regeneration DeepSpeedPlugin](https://github.com/huggingface/accelerate/blob/3499cf25aa431ed3eeb7969f1d9b501ef20d0911/src/accelerate/accelerator.py#L284-L292)
[source code: HfTrainerDeepSpeedConfig  -> HfDeepSpeedConfig ](https://github.com/huggingface/accelerate/blob/3499cf25aa431ed3eeb7969f1d9b501ef20d0911/src/accelerate/utils/dataclasses.py#L600)

which may lead to errors similar to https://github.com/microsoft/DeepSpeed/issues/3733, https://github.com/hiyouga/LLaMA-Factory/issues/286. If this situation occurs, it will be difficult to identify the problem. It feels that using the ```LOCAL_RANK``` environment variable may be a better approach.

### CODE REVIEW

**Comment by lvwerra** [comment]
Maybe @muellerzr would know if this is a viable alternative? Otherwise I can also try to double check myself.

File: trl/models/modeling_base.py
Comment (muellerzr) [inline trl/models/modeling_base.py:400]: This whole thing can be replaced with `state.local_process_index` unless `xpu` is needed.

Comment (muellerzr) [reply trl/models/modeling_base.py:400]: I'd say that can be in another PR then just to keep things isolated a bit :) LG2M then!

**Review by muellerzr — APPROVED** [review]

**Review by lvwerra — APPROVED** [review]

**Comment by muellerzr** [comment]
@billvsme ideally can you run it again and verify this fixes the problem with `PartialState()`? (want to make sure we're not in the same position)

**Comment by billvsme** [comment]
@muellerzr Yes, solved. (I changed a annotation after you approved and submitted it once)

**Pull request merged**
